lib/jd_pay/util.rb

```diff
@@ -46,6 +46,34 @@ def build_pay_form(url, form_attributes)
             </body>
         </html>".gsub(/>[[:space:]]+/, ">")
       end
+
+      def stringify_keys(hash)
+        new_hash = {}
+        hash.each do |key, value|
+          new_hash[(key.to_s rescue key) || key] = value
+        end
+        new_hash
+      end
+
+      def decrypt_and_drop_empty(hash)
+        decrypted = {}
+        new_hash = {}
+        hash.each do |k, v|
+          # Decrypt all valid value
+          if %w(tradeNum amount currency tradeTime note status).include? k
+            v = JdPay::Des.decrypt_3des(v) unless v == ''
+            decrypted[k] = v
+            # According to the doc, any empty value should be dropped.
+            next if v == ''
+            new_hash[k] = v
+          end
+        end
+        return decrypted, new_hash
+      end
+
+      def params_to_string(params)
```

Comment:
Reinventing the wheel. rename the `to_uri` ?
[Hash#to_query](https://github.com/rails/rails/blob/375a4143cf5caeb6159b338be824903edfd62836/activesupport/lib/active_support/core_ext/object/to_query.rb#L76)

Reply:
Good point. 👍

Reply:
you misunderstood me on this.
rename the `to_uri` using the `to_query`, there is a method named `to_uri`.